Rules B009 cause autofix error

Ruff 0.0.286 (latest changes from main branch)

```
ruff  *.py --select ALL,NURSERY --no-cache --fix
```

file content:
```
print(getattr(1, "real"))
```

error:
```
error: Autofix introduced a syntax error. Reverting all changes.

This indicates a bug in `ruff`. If you could open an issue at:

    https://github.com/astral-sh/ruff/issues/new?title=%5BAutofix%20error%5D

...quoting the contents of `Attributes3841.py`, the rule codes B009, along with the `pyproject.toml` settings and executed command, we'd be very appreciative!

Attributes3841.py:1:1: T201 `print` found
Attributes3841.py:1:1: D100 Missing docstring in public module
Attributes3841.py:1:1: CPY001 Missing copyright notice at top of file
Attributes3841.py:1:7: B009 Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

```
[Attributes3841.py.zip](https://github.com/astral-sh/ruff/files/12467952/Attributes3841.py.zip)

--- a/crates/ruff/resources/test/fixtures/flake8_bugbear/B009_B010.py
+++ b/crates/ruff/resources/test/fixtures/flake8_bugbear/B009_B010.py
@@ -1,7 +1,7 @@
 """
 Should emit:
-B009 - Line 19, 20, 21, 22, 23, 24
-B010 - Line 40, 41, 42, 43, 44, 45
+B009 - Lines 19-31
+B010 - Lines 40-45
 """
 
 # Valid getattr usage
@@ -24,6 +24,12 @@ getattr(foo, r"abc123")
 _ = lambda x: getattr(x, "bar")
 if getattr(x, "bar"):
     pass
+getattr(1, "real")
+getattr(1., "real")
+getattr(1.0, "real")
+getattr(1j, "real")
+getattr(True, "real")
+
 
 # Valid setattr usage
 setattr(foo, bar, None)

--- a/crates/ruff/src/rules/flake8_bugbear/rules/getattr_with_constant.rs
+++ b/crates/ruff/src/rules/flake8_bugbear/rules/getattr_with_constant.rs
@@ -79,21 +79,22 @@ pub(crate) fn getattr_with_constant(
 
     let mut diagnostic = Diagnostic::new(GetAttrWithConstant, expr.range());
     if checker.patch(diagnostic.kind.rule()) {
-        let fixed = if matches!(
-            obj,
-            Expr::Constant(ast::ExprConstant {
-                value: Constant::Int(_),
-                ..
-            })
-        ) {
-            // Attribute accesses on `int` literals must be parenthesized.
-            // e.g., `getattr(1, "real")` becomes `(1).real`
-            format!("({}).{}", checker.locator().slice(obj), value)
-        } else {
-            format!("{}.{}", checker.locator().slice(obj), value)
-        };
-
-        diagnostic.set_fix(Fix::suggested(Edit::range_replacement(fixed, expr.range())));
+        diagnostic.set_fix(Fix::suggested(Edit::range_replacement(
+            if matches!(
+                obj,
+                Expr::Constant(ast::ExprConstant {
+                    value: Constant::Int(_),
+                    ..
+                })
+            ) {
+                // Attribute accesses on `int` literals must be parenthesized, e.g.,
+                // `getattr(1, "real")` becomes `(1).real`.
+                format!("({}).{}", checker.locator().slice(obj), value)
+            } else {
+                format!("{}.{}", checker.locator().slice(obj), value)
+            },
+            expr.range(),
+        )));
     }
     checker.diagnostics.push(diagnostic);
 }

--- a/crates/ruff/src/rules/flake8_bugbear/rules/getattr_with_constant.rs
+++ b/crates/ruff/src/rules/flake8_bugbear/rules/getattr_with_constant.rs
@@ -1,7 +1,6 @@
-use ruff_python_ast::{self as ast, Constant, Expr};
-
 use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit, Fix};
 use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast, Constant, Expr};
 use ruff_python_stdlib::identifiers::{is_identifier, is_mangled_private};
 use ruff_text_size::Ranged;
 
@@ -80,10 +79,21 @@ pub(crate) fn getattr_with_constant(
 
     let mut diagnostic = Diagnostic::new(GetAttrWithConstant, expr.range());
     if checker.patch(diagnostic.kind.rule()) {
-        diagnostic.set_fix(Fix::suggested(Edit::range_replacement(
-            format!("{}.{}", checker.locator().slice(obj), value),
-            expr.range(),
-        )));
+        let fixed = if matches!(
+            obj,
+            Expr::Constant(ast::ExprConstant {
+                value: Constant::Int(_),
+                ..
+            })
+        ) {
+            // Attribute accesses on `int` literals must be parenthesized.
+            // e.g., `getattr(1, "real")` becomes `(1).real`
+            format!("({}).{}", checker.locator().slice(obj), value)
+        } else {
+            format!("{}.{}", checker.locator().slice(obj), value)
+        };
+
+        diagnostic.set_fix(Fix::suggested(Edit::range_replacement(fixed, expr.range())));
     }
     checker.diagnostics.push(diagnostic);
 }

